app/ui-react/syndesis/src/modules/connections/components/Connections.tsx

```diff
@@ -78,16 +79,11 @@ export class Connections extends React.Component<IConnectionsProps> {
                               };
 
                               const configurationRequired =
-                                c.board &&
-                                (c.board!.notices ||
-                                  c.board!.warnings ||
-                                  c.board!.errors)! > 0;
+                                c.board && isConfigurationRequired(c.board);
 
-                              const isTechPreview =
-                                c.connector! && c.connector!.metadata!
-                                  ? c.connector!.metadata!['tech-preview'] ===
-                                    'true'
-                                  : false;
+                              const techPreview = c.connector
```

Comment:
Can't this be treated like [this](https://github.com/syndesisio/syndesis-react/pull/409/files#diff-c8c891bfe81f60cb81839bf4031cb8c7R19)?

Reply:
`techPreview` relied on `IUIStep` based on the feedback from before, which couldn't be access from those API helper files since it was within `modules/integrations/components/editor/interfaces.ts`. If we only accept `Connector` then it should be fine and I can get rid of the generics stuff as well.